nucypher/blockchain/eth/actors.py

```diff
@@ -659,3 +676,367 @@ def request_evaluation(self, evidence):
+            # TODO: More formal check for wallet here?
+            # With devices, the last account is is always an unsed one.
+            new_account = self.blockchain.client.accounts[-1]
+        else:
+            if not password:
+                raise self.ConfigurationError("No password supplied to create new staking account.")
+            new_account = self.blockchain.client.new_account(password=password)
+        self.__accounts.append(new_account)
+        staker = Staker(blockchain=self.blockchain, is_me=True, checksum_address=new_account)
+        return staker
+
+    def create_worker_configuration(self,
+                                    staking_address: str,
+                                    worker_address: str,
+                                    password: str,
+                                    **configuration):
+
+        """Generates a worker JSON configuration file for a given staking address."""
+        from nucypher.config.characters import UrsulaConfiguration
+        worker_configuration = UrsulaConfiguration.generate(checksum_address=staking_address,
+                                                            worker_address=worker_address,
+                                                            password=password,
+                                                            config_root=self.config_root,
+                                                            federated_only=False,
+                                                            provider_uri=self.blockchain.provider_uri,
+                                                            **configuration)
+        return worker_configuration
+
+    #
+    # Actions
+    #
+
+    def set_worker(self, staker_address: str, worker_address: str, password: str = None):
+        self.attach_transacting_power(checksum_address=staker_address, password=password)
+        staker = self.get_active_staker(address=staker_address)
+        result = self.staking_agent.set_worker(staker_address=staker.checksum_address,
+                                               worker_address=worker_address)
+
+        self.to_configuration_file(override=True)
+        return result
+
+    def __prepare_new_staker(self, staker: Staker, amount: NU):
+        """Creates a new ethereum account, then transfers it ETH and NU for staking."""
+        if self.funding_tokens < amount:
+            delta = amount - staker.token_balance
+            raise self.ConfigurationError(f"{self.funding_account} Insufficient NU (need {delta} more).")
+        if not self.funding_eth:
+            raise self.ConfigurationError(f"{self.funding_account} has no ETH")
+
+        # Transfer NU
+        self.funding_power.activate()
+
+        # Send new staker account ETH
+        tx = {'to': staker.checksum_address,
+              'from': self.funding_account,
+              'value': self.eth_funding}
+        _ether_transfer_receipt = self.blockchain.client.send_transaction(transaction=tx)
```

Comment:
Ok, I think we have a problem here.
`send_transaction` doesn't actually wait for receipt, instead it returns a transaction hash (as a bytestring). If the blockchain is slow enough (not automine!), the transaction won't get included by the time we try to submit the next one, it'll have a wrong nonce and fail.
We need to wait for receipt either in `send_transaction`, or here, such as:
```
        _txhash = self.blockchain.client.send_transaction(transaction=tx)
        _ether_transfer_receipt = self.blockchain.client.wait_for_receipt(_txhash, TIMEOUT)
```